beta-src/src/components/ui/WDMain.tsx

```diff
@@ -1,13 +1,55 @@
 import * as React from "react";
+import {
+  fetchGameData,
+  fetchGameOverview,
+  gameApiSliceActions,
+  gameOverview,
+  userActivity,
+} from "../../state/game/game-api-slice";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import debounce from "../../utils/debounce";
 import WDUI from "./WDUI";
 
 const WDMapController = React.lazy(() => import("../map/WDMapController"));
 
 const WDMain: React.FC = function (): React.ReactElement {
+  const dispatch = useAppDispatch();
+  const { season, year, processTime, makeNewCall } =
+    useAppSelector(userActivity);
+  const {
+    season: newSeason,
+    year: newYear,
+    processTime: newProcessTime,
+    gameID,
+    user: {
+      member: { countryID },
+    },
+  } = useAppSelector(gameOverview);
+  if (makeNewCall) {
+    dispatch(fetchGameOverview({ gameID: String(gameID) }));
+  }
+  if (
+    processTime &&
+    (season !== newSeason || year !== newYear || processTime !== newProcessTime)
+  ) {
+    dispatch(
+      fetchGameData({ gameID: String(gameID), countryID: String(countryID) }),
+    );
+  }
+  const activityHandler = debounce(() => {
+    dispatch(
+      gameApiSliceActions.updateUserActivity({
+        // eslint-disable-next-line no-bitwise
+        lastActive: (Date.now() / 1000) | 0,
+      }),
+    );
+  }, 500);
   return (
     <React.Suspense fallback={<div>Loading...</div>}>
-      <WDMapController />
-      <WDUI />
+      <div onMouseMove={activityHandler[0]} onClickCapture={activityHandler[0]}>
```

Comment:
Could we move this logic to a wrapper component and just move all the logic in there? Just to keep WDMain clean.

Reply:
Hmm, WDMain is the wrapper, but is there a reason why we want to keep this file clean?

Reply:
To me seems like this is all logic can be moved into a component and used as a wrapper in here. WDMain should only be holding UI components.

```
<WDWrappperLogic>
   <WDMapController />
   <WDUI />
</WDWrappperLogic>
 ```